tests/helper/sampler_test.cpp

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+#include "common.hpp"
+#include <sampler.hpp>
+#include <thread>
+
+namespace dds {
+
+std::atomic<int> picked = 0;
+
+void count_picked(dds::sampler &sampler, int iterations)
+{
+    for (int i = 0; i < iterations; i++) {
+        auto is_pick = sampler.get();
+        if (is_pick != std::nullopt) {
+            picked++;
+        }
+    }
+}
+
+int run_sample(dds::sampler &sampler)
```

Comment:
This looks overly complex for simple cases in which the sampler will just work as expected when performing sequential calls, since it's not affected by time or by number of threads at all.

I'm not suggesting you shouldn't have multithreaded test cases, but I am suggesting that you should have single-threaded cases as well.